helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/constraints/ConstraintBasedAlgorithmFactory.java

```diff
@@ -61,21 +61,34 @@ public static RebalanceAlgorithm getInstance(
             new ReplicaActivateConstraint(), new NodeMaxPartitionLimitConstraint(),
             new ValidGroupTagConstraint(), new SamePartitionOnInstanceConstraint());
 
-    int evennessPreference =
-        preferences.getOrDefault(ClusterConfig.GlobalRebalancePreferenceKey.EVENNESS, 1);
-    int movementPreference =
-        preferences.getOrDefault(ClusterConfig.GlobalRebalancePreferenceKey.LESS_MOVEMENT, 1);
+    int evennessPreference = 1;
+    int movementPreference = 1;
+    if (preferences.containsKey(ClusterConfig.GlobalRebalancePreferenceKey.EVENNESS) && preferences
+        .containsKey(ClusterConfig.GlobalRebalancePreferenceKey.EVENNESS)) {
+      evennessPreference = preferences.get(ClusterConfig.GlobalRebalancePreferenceKey.EVENNESS);
+      movementPreference =
+          preferences.get(ClusterConfig.GlobalRebalancePreferenceKey.LESS_MOVEMENT);
+    }
+
+    boolean forceBaselineConverge = preferences
+        .getOrDefault(ClusterConfig.GlobalRebalancePreferenceKey.FORCE_BASELINE_CONVERGE, 0)
+        > 0;
+    if (forceBaselineConverge) {
+      MODEL.put(PartitionMovementConstraint.class.getSimpleName(), 0.001f);
```

Comment:
I think it would be better not to touch this weight. Unlike BaselineInfluenceConstraint, PartitionMovementConstraint is used in the baseline calculation too. So if you modify the weight here, then the baseline will be over evened. This may cause a large partition shuffling.

What we can do should be modifying BaselineInfluenceConstraint **after** the preference has been applied to the the constraint weight.

Reply:
And I guess set its weight to (10000 * PartitionMovementConstraint weight * LESS_MOVEMENT preference factor) should be enough and 0.001f maybe not required.

Reply:
For the cluster that motivates this change, an extremely small number like 0.001 is required to achieve better baseline evenness than CrushED. The quality of the baseline is only controlled by this factor. 

Note that we cannot set the 0.001 with preferences, since they must be integers. This part is where I find this approach having a hard time creating something elegant, unless we use system properties to overwrite.

Reply:
Let's discuss offline, what you want to do is 2 features instead of one. This could be concerning. Since this forceBaselineConverge option is not exactly what it claims to be.

Reply:
That's exactly my concern. Let's discuss.

Reply:
Conclusion: for this PR, I'm removing this line. Another solution will be introduced later if necessary.